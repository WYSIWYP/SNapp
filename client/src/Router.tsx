import React from 'react';
import {Router as ReachRouter} from '@reach/router';
import Convert from './pages/Convert';
import NotFound from './pages/NotFound';
import GlobalStateProvider from './contexts/Global';
import Global from './components/global/Global';
import Menu from './pages/Menu';

const Router: React.FC = () => {
    return (
        <GlobalStateProvider>
            <Global />
            <ReachRouter>
                <Menu path="/" />
                <Convert path="/convert" />
                <NotFound path="/*" />
            </ReachRouter>
        </GlobalStateProvider>
    );
}

export default Router;
