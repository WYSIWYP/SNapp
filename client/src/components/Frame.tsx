import React, {CSSProperties, useState, useRef, useContext, useEffect} from 'react';

type Props = {
    header: string,
    fontSize?: number,
    SideMenu?: React.ReactNode,
    ShowSideMenu?: boolean,
};

const Frame: React.FC<Props> = ({header,fontSize,SideMenu,ShowSideMenu,children}) => {
    

    return (
        <div style={styles.appContainer}>
            <div style={{...styles.header, fontSize: fontSize || 45}}>
                {header}
                {/* <div style={styles.headerLeft}>
                    {headerLeft}
                </div>
                <div style={styles.headerRight}>
                    {headerRight}
                </div> */}
            </div>

            
            <div style={styles.sideBar}>
                    <div style={styles.sideBarTop}>
                        Export 
                        Import
                        Close X
                    </div>
                    <div style={styles.sideBarContent}>
                        <div style={styles.line}>
                            <div style={styles.name}>Staff Scale</div>
                            <div style={styles.option}>Medium </div>
                        </div>

                        <div style={styles.line}>
                            <div style={styles.name}>Horizontal Spacing</div>
                            <div style={styles.option}>10 px </div>
                        </div>

                        <div style={styles.line}>
                            <div style={styles.name}>Vertical Spacing</div>
                            <div style={styles.option}> 20 px </div>
                        </div>


                    </div>
            </div>


            <div style={styles.page}>
                {children}

                
            </div>
        </div>
    );
};

const styleMap = {
    name:{
        position:'relative',
        width: '50%',
        paddingTop: '6%',
    },
    option:{
        paddingTop: '5%',
        background: '#F5F5F5',
        border: '1px solid #6F6F6F',
        boxSizing: 'border-box',
        borderRadius: '10px',
        position:'relative',
        width: '50%',
        textAlign:'center',

    },
    line:{

        margin: '30px 0px',
        justifyContent: 'center',
        alignItems: 'center',
        display: 'flex',
        position:'relative',
        height:'50px',


    },
    sideBarContent:{
        padding: '0 20px',
        position:'relative',
        marginTop: '40px',
    },
    appContainer: {
        overflow: 'hidden',
    },
    header: {
        // fontFamily: 'open sans',
        fontStyle: 'normal',
        fontWeight: 'normal',
        fontSize: '42px',
        alignItems: 'center',
        textAlign: 'center',
        letterSpacing: '0.15em',
        left: '50%',
        top:'0px',
        height: '80px',
        transform: 'translate(-50%,0px)',
        backgroundColor: '#31B7D6',
        minWidth: '600px',
        position: 'absolute',
    },
   
    page: {
        top: '100px',
        height: 'calc(100% - 100px)',
    },
    sideBar:{
        top: '80px',
        height: 'calc(100% - 100px)',

        borderLeft: '1px solid #6F6F6F',
        boxSizing: 'border-box',
        position:'relative',

        width:'20%',
        float:'right',

    },

    sideBarTop:{
        justifyContent: 'center',
        alignItems: 'center',
        borderBottom: '1px solid #6F6F6F',
        boxSizing: 'border-box',
        height: '80px',
        display:'flex',
        color:'#31B7D6',
        fontSize: '25px',
        fontWeight:'bold',
        cursor: 'pointer',
        position: 'relative',
        width: 'auto',
    }


} as const;
const styles: Record<keyof typeof styleMap, CSSProperties> = styleMap;

export default Frame;
