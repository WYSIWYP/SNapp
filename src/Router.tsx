import React from 'react';
import {Router as ReachRouter} from '@reach/router';
import Convert from './pages/Convert';
import NotFound from './pages/NotFound';
import GlobalStateProvider from './contexts/Global';
import Global from './components/global/Global';
import Menu from './pages/Menu';

const Router: React.FC = () => {
    return (
        <GlobalStateProvider>
            <ReachRouter>
                <Menu path="/" />
                <Convert path="/convert" />
                <NotFound path="/*" />
            </ReachRouter>
            <Global />
        </GlobalStateProvider>
    );
};

export default Router;
