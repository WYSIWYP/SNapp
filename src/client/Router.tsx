import React from 'react';
import {Router as ReachRouter} from '@reach/router';
import Menu from './pages/Convert';
import NotFound from './pages/NotFound';
import GlobalStateProvider from './contexts/Global';
import Global from './components/global/Global';

const Router: React.FC = () => {
    return (
        <GlobalStateProvider>
            <Global />
            <ReachRouter>
                <Menu path="/" />
                <NotFound path="/*" />
            </ReachRouter>
        </GlobalStateProvider>
    );
}

export default Router;
