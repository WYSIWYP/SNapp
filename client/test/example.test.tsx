import React from 'react';
import {shallow} from 'enzyme';

import Frame from '../src/components/Frame';
import Router from '../src/Router';

it('renders correctly ...', () => {
    let tree = shallow(<Frame header="Header Text"/>);
    expect(tree).toMatchSnapshot();

    tree = shallow(<Router/>);
    expect(tree).toMatchSnapshot();

});
