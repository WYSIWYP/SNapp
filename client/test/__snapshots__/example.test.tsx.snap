// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly ... 1`] = `
<div
  style={
    Object {
      "overflow": "hidden",
    }
  }
>
  <div
    id="header"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#31B7D6",
        "fontSize": 45,
        "fontStyle": "normal",
        "fontWeight": "normal",
        "height": "80px",
        "left": "50%",
        "letterSpacing": "0.15em",
        "minWidth": "600px",
        "paddingTop": "8px",
        "position": "absolute",
        "textAlign": "center",
        "top": "0px",
        "transform": "translate(-50%,0px)",
      }
    }
  >
    Header Text
  </div>
  <div
    style={
      Object {
        "height": "calc(100% - 80px)",
        "overflow": "auto",
        "top": "80px",
      }
    }
  />
  <div
    className="animated close"
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.7)",
        "height": "calc(100% - 80px)",
        "left": "auto",
        "overflowX": "hidden",
        "overflowY": "scroll",
        "right": "0px",
        "top": "80px",
      }
    }
  />
</div>
`;

exports[`renders correctly ... 2`] = `
<GlobalStateProvider>
  <Global />
  <Router>
    <Menu
      path="/"
    />
    <Convert
      path="/convert"
    />
    <NotFound
      path="/*"
    />
  </Router>
</GlobalStateProvider>
`;
